purchases/src/test/java/com/revenuecat/purchases/PurchaserInfoTest.kt

```diff
@@ -235,4 +236,48 @@ class PurchaserInfoTest {
         assertThat(x.originalPurchaseDate).isNull()
     }
 
+    @Test
+    fun `Non subscription transactions is empty if there are no non_subscriptions`() {
+        val jsonObject = JSONObject(Responses.validEmptyPurchaserResponse)
+        val x = jsonObject.buildPurchaserInfo()
+
+        assertThat(x.nonSubscriptionTransactions).isEmpty()
+    }
+
+    @Test
+    fun `Non subscription transactions is correctly created`() {
+        val jsonObject = JSONObject(Responses.validFullPurchaserResponse)
+        val x = jsonObject.buildPurchaserInfo()
+
+        assertThat(x.nonSubscriptionTransactions).isNotEmpty
+        assertThat(x.nonSubscriptionTransactions.size).isEqualTo(2)
+
+        val oneTimePurchaseTransactions = x.nonSubscriptionTransactions["onetime_purchase"]
+            ?: error("onetime_purchase transactions are null")
+        assertThat(oneTimePurchaseTransactions.size).isEqualTo(2)
+        oneTimePurchaseTransactions.forEach {
+            assertThat(it.productId).isEqualTo("onetime_purchase")
+        }
+
+        val consumableTransactions = x.nonSubscriptionTransactions["consumable"]
+            ?: error("consumable transactions are null")
+        assertThat(consumableTransactions.size).isEqualTo(2)
+        consumableTransactions.forEach {
+            assertThat(it.productId).isEqualTo("consumable")
+        }
+
+        assertThat((oneTimePurchaseTransactions + consumableTransactions)
+            .distinctBy { it.transactionId }.size).isEqualTo(4)
```

Comment:
I thought that making sure the 4 transactions are different is enough. I thought comparing one by one was overkill since they come from a JSON file.